File: app/src/main/java/com/kunzisoft/keepass/model/Entry.java
package com.kunzisoft.keepass.model;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.ArrayList;
import java.util.List;

public class Entry implements Parcelable {

    private String title;
    private String username;
    private String password;
    private String url;
    private List<Field> customFields;

    public Entry() {
        this.title = "";
        this.username = "";
        this.password = "";
        this.url = "";
        this.customFields = new ArrayList<>();
    }

    protected Entry(Parcel in) {
        title = in.readString();
        username = in.readString();
        password = in.readString();
        url = in.readString();
        //noinspection unchecked
        customFields = in.readArrayList(Field.class.getClassLoader());
    }

    public static final Creator<Entry> CREATOR = new Creator<Entry>() {
        @Override
        public Entry createFromParcel(Parcel in) {
            return new Entry(in);
        }

        @Override
        public Entry[] newArray(int size) {
            return new Entry[size];
        }
    };

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public List<Field> getCustomFields() {
        return customFields;
    }

    public void addCustomField(Field field) {
        this.customFields.add(field);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(title);
        parcel.writeString(username);
        parcel.writeString(password);
        parcel.writeString(url);
        parcel.writeArray(customFields.toArray());
    }
}

File: app/src/main/java/com/kunzisoft/keepass/model/Entry.kt
package com.kunzisoft.keepass.model

import android.os.Parcel
import android.os.Parcelable

import java.util.ArrayList

class Entry : Parcelable {

    var title: String = ""
    var username: String = ""
    var password: String = ""
    var url: String = ""
    var customFields: MutableList<Field> = ArrayList()

    constructor()

    private constructor(parcel: Parcel) {
        title = parcel.readString()
        username = parcel.readString()
        password = parcel.readString()
        url = parcel.readString()
        parcel.readList(customFields, Field::class.java.classLoader)
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeString(title)
        parcel.writeString(username)
        parcel.writeString(password)
        parcel.writeString(url)
        parcel.writeArray(customFields.toTypedArray())
    }

    companion object {

        @JvmField
        val CREATOR: Parcelable.Creator<Entry> = object : Parcelable.Creator<Entry> {
            override fun createFromParcel(parcel: Parcel): Entry {
                return Entry(parcel)
            }

            override fun newArray(size: Int): Array<Entry?> {
                return arrayOfNulls(size)
            }
        }
    }
}
